raft/raft_test.go

```diff
@@ -288,33 +288,80 @@ func TestProgressPaused(t *testing.T) {
 }
 
 func TestLeaderElection(t *testing.T) {
```

Comment:
OK, refactored all the tests I touched to use two separate test functions instead of one with a loop.